kafka/src/test/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisherTest.java

```diff
@@ -326,12 +325,12 @@ public void testSendMessageWithKafkaTransactionRollback() {
         return producerFactory;
     }
 
-    private KafkaPublisher<?, ?> buildPublisher(String topic) {
+    private KafkaPublisher<String, byte[]> buildPublisher(String topic) {
         DefaultKafkaMessageConverter messageConverter =
                 DefaultKafkaMessageConverter.builder()
                                             .serializer(XStreamSerializer.builder().build())
                                             .build();
-        KafkaPublisher<?, ?> testSubject = KafkaPublisher.<String, byte[]>builder()
+        KafkaPublisher<String, byte[]> testSubject = KafkaPublisher.<String, byte[]>builder()
```

Comment:
I will suggest to use the `testSubject` declared on top of the test class in order to avoid confusion.

Reply:
Due to the `EmbeddedKafkaBroker` being used, with a distinct topic per test, we need to build the `testSubject` in every test separately. I would also prefer to have it on top level, but I cannot find a quick/simple way around this.

Reply:
Good point still...I've renamed it to `kafkaPublisher` in this method, as during construction in this function it is not yet the actual `testSubject`. Hope this resolves some of the confusion.